integration-libs/cpq-quote/root/components/cpq-quote-discount-tbody/cpq-quote.component.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, Optional, OnDestroy, OnInit } from '@angular/core';
+import { CartItemContext, OrderEntry } from '@spartacus/cart/base/root';
+import { EMPTY, Observable, Subscription } from 'rxjs';
+
+@Component({
+  selector: 'cx-cpq-quote',
+  templateUrl: './cpq-quote.component.html',
+})
+export class CpqQuoteDiscountComponent implements OnInit, OnDestroy {
+  quoteDiscountData: OrderEntry;
+  private subscription: Subscription;
+  constructor(@Optional() protected cartItemContext: CartItemContext) {}
```

Comment:
MAJOR: lets use inject instead of constructor.

For more reasons why we introduced this recommendation, feel free to check the Architectural Decision Document "[Use inject() function instead of constructor for Dependency Injection](https://wiki.one.int.sap/wiki/display/spar/Use+inject%28%29+function+instead+of+constructor+for+Dependency+Injection)"
You can find the above mentioned guideline also in our Spartacus [Coding Guidelines](https://wiki.one.int.sap/wiki/display/spar/Coding+Guidelines#CodingGuidelines-Useinject()functioninsteadofconstructorforDependencyInjection) wiki.